File: crescer/java/trabalhos/tcc/api/src/main/java/br/com/cwi/crescer/api/service/security/SecurityUser.java
package br.com.cwi.crescer.api.service.security;

import br.com.cwi.crescer.api.model.UserAccount;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.List;
import java.util.stream.Collectors;

public class SecurityUser implements UserDetails {
    private Long id;
    private String password;
    private String email;
    private List<SimpleGrantedAuthority> permissions;

    public SecurityUser (UserAccount user){
        this.id= user.getUserId();
        this.email=user.getEmail();
        this.password= user.getPassword();
        this.permissions = user.getPermissionList()
                .stream()
                .map(permission -> new SimpleGrantedAuthority("ROLE_"+ permission.getPermissionName()))
                .collect(Collectors.toList());
    }

    public Long getId (){
        return this.id;
    }
    @Override
    public List<SimpleGrantedAuthority> getAuthorities() {
        return this.permissions;
    }

    @Override
    public String getPassword() {
        return this.password;
    }

    @Override
    public String getUsername() {
        return this.email;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: crescer/trabalhos/healthCare.api/src/main/kotlin/com/healthCare/api/service/user/SecurityUser.kt
package com.healthCare.api.service.user

import com.healthCare.api.model.Permission
import com.healthCare.api.model.UserAccount
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails
import java.util.stream.Collectors

class SecurityUser(user: UserAccount) : UserDetails {
    val id: Long
    private val password: String
    private val email: String
    private val permissions: List<SimpleGrantedAuthority>

    init {
        id = user.id
        email = user.email
        password = user.password
        permissions = user.permissionList
            .stream()
            .map { (_, permissionName): Permission ->
                SimpleGrantedAuthority(
                    "ROLE_$permissionName"
                )
            }
            .collect(Collectors.toList())
    }

    override fun getAuthorities(): List<SimpleGrantedAuthority> {
        return permissions
    }

    override fun getPassword(): String {
        return password
    }

    override fun getUsername(): String {
        return email
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }
}